src/Constraint/Constraint.php

```diff
@@ -171,6 +171,77 @@ public function versionCompare($a, $b, $operator, $compareBranches = false)
         return version_compare($a, $b, $operator);
     }
 
+    public function compile($otherOperator) {
+        $isBranch = 'dev-' === substr($this->version, 0, 4);
+
+        if (self::OP_NE === $this->operator) {
+            if ($isBranch) {
+                if ($otherOperator === self::OP_EQ) {
+                    return 'false';
+                }
+
+                return 'true';
+            }
+
+            if ($otherOperator === self::OP_EQ) {
+                return sprintf('!$b && version_compare($v, \'%s\', \'!=\')', $this->version);
+            }
+
+            return 'true';
+        } elseif (self::OP_EQ === $this->operator) {
+            if ($isBranch) {
+                if (in_array($otherOperator, array(self::OP_LT, self::OP_GT, self::OP_NE))) {
+                    return 'false';
+                }
+
+                return sprintf('$b && $v === \'%s\'', $this->version);
```

Comment:
use `var_export($this->version, true)` to generate the PHP string representation rather than `sprintf('\'%s\'', $this->version)`. Otherwise, you might face issues in some cases (potentially even allow arbitrary code injection in the composer solver). AFAIK, single quotes are not forbidden in versions when using `dev-*` versions for instance.

Reply:
thanks